### ISSUE TITLE
Change detector no longer receiving events after restarting docker on QNap

### ISSUE THREAD

**Issue by okachobi** [cantino/huginn#1809]
I run huginn on docker on my qnap server and had a working scenario that I used for monitoring my sauna temperature and turning on the light when it got to the proper temperature. (also sends an email and sets another lamp to a flashing mode).  And it can turn off the light when the temperature drops below 50C.

2 days ago something changed- that is when the last even reached the second agent in the chain. The first agent simply polls the temperature service and gets back something like:

{ "status": "ok", "sender": 5, "devtype": 1, "SW1": 0 , "H": 14.60 , "SW2": 1 , "T": 57.80 }

And changes it to something like:

{"temperature"=>47.8, "humidity"=>14.60, "switch_1"=>0, "switch_2"=>1}

Everything was working great, then I went to use the sauna today and the light didn't turn on.  I checked and the first agent in the graph is generating events, but the 2nd agent, a change detector, is showing it never receives events.  It shows that the first agent is the source is listens for.  I deleted that relationship and added it again just in case...and still no event propagate to it.  I see the first event running on its 2 minute schedule and the results are being successfully parsed, but I just see...

RESULT 109                                                                                                                                                                                                                                                    
processname:foreman groupname:foreman pid:419 channel:stdout                                                                                                                                                                                                  
20:05:54 jobs.1 | Queuing schedule for every_1m                                                                                                                                                                                                               
READY                                                                                                                                                                                                                                                         
RESULT 105                                                                                                                                                                                                                                                    
processname:foreman groupname:foreman pid:419 channel:stdout                                                                                                                                                                                                  
20:05:54 jobs.1 | Queuing event propagation                                                                                                                                                                                                                   
READY                                                                                                                                                                                                                                                         
RESULT 109                                                                                                                                                                                                                                                    
processname:foreman groupname:foreman pid:419 channel:stdout                                                                                                                                                                                                  
20:06:51 jobs.1 | Queuing schedule for every_2m                                                                                                                                                                                                               
READY                                                                                                                                                                                                                                                         
RESULT 109                                                                                                                                                                                                                                                    
processname:foreman groupname:foreman pid:419 channel:stdout                                                                                                                                                                                                  
20:06:54 jobs.1 | Queuing schedule for every_1m                                                                                                                                                                                                               
READY                                                                                                                                                                                                                                                         
RESULT 105                                                                                                                                                                                                                                                    
processname:foreman groupname:foreman pid:419 channel:stdout                                                                                                                                                                                                  
20:06:54 jobs.1 | Queuing event propagation                                                                                                                                                                                                                   
READY                                                                                                                                                                                                                                                         
RESULT 140                                                                                                                                                                                                                                                    
processname:foreman groupname:foreman pid:419 channel:stdout                                                                                                                                                                                                  
20:07:00 jobs.1 | Agent#10: Fetching http://192.168.0.20:3128/lightgw/5/status                                                                                                                                                                                
READY                                                                                                                                                                                                                                                         
RESULT 118                                                                                                                                                                                                                                                    
processname:foreman groupname:foreman pid:419 channel:stdout                                                                                                                                                                                                  
20:07:00 jobs.1 | Agent#10: Extracting json at T: [43.5]                                                                                                                                                                                                      
READY                                                                                                                                                                                                                                                         
RESULT 118                                                                                                                                                                                                                                                    
processname:foreman groupname:foreman pid:419 channel:stdout                                                                                                                                                                                                  
20:07:00 jobs.1 | Agent#10: Extracting json at H: [18.4]                                                                                                                                                                                                      
READY                                                                                                                                                                                                                                                         
RESULT 117                                                                                                                                                                                                                                                    
processname:foreman groupname:foreman pid:419 channel:stdout                                                                                                                                                                                                  
20:07:00 jobs.1 | Agent#10: Extracting json at SW1: [0]                                                                                                                                                                                                       
READY                                                                                                                                                                                                                                                         
RESULT 117                                                                                                                                                                                                                                                    
processname:foreman groupname:foreman pid:419 channel:stdout                                                                                                                                                                                                  
20:07:00 jobs.1 | Agent#10: Extracting json at SW2: [1]                                                                                                                                                                                                       
READY                                                                                                                                                                                                                                                         
RESULT 217                                                                                                                                                                                                                                                    
processname:foreman groupname:foreman pid:419 channel:stdout                                                                                                                                                                                                  
20:07:01 jobs.1 | Agent#10: Storing new parsed result for 'Sauna Temperature Reader': {"temperature"=>43.5, "humidity"=>18.4, "switch_1"=>0, "switch_2"=>1}                                                                                                   
READY

There is never any indication that the change detector is being executed.  I clear its memory and the value is never being set.

Do I have a data corruption issue in my database?  I've changed nothing, I just rebooted the NAS and according to the "last input" date, this failed before that reboot.  Everything else on it seems to be working.  I have the mysql folder on a map to the host, as well as /etc/localtime.  I see in the app/log/ that the selects are occurring, but it just doesn't seem to be propagating.  Is there an easy way to clear the MySQL tables and reload the scenario?

[notify-me-when-sauna-heats-up.txt](https://github.com/cantino/huginn/files/602469/notify-me-when-sauna-heats-up.txt)

**Comment by cantino** [comment]
Hey @okachobi. Fun use case! So you did see this working correctly for a while, and then it stopped, without any changes on your part? Is there anything in the log tab when viewing the individual website or change detector agents?

**Comment by okachobi** [comment]
Yes, this has worked for many weeks over the summer and to my knowledge, nothing had changed.  I did have to restart the arduino that ran the temperature sensor- it was not responding to the 433hz radio network, so my web gateway was returning 500 errors.  But  after I did that, I started seeing the events in the first agent that reads the http request, but they did not get routed to the change detector agent.

Oddly, just now I went to test again to look at the log tab, and the events are now propagating properly.  So I'm confused why they weren't before....but it appears to be working now.  So it has corrected itself.  I'm not sure what the problem was before.

btw- the code for the arduino/jeenode network and http gateway is [here](https://github.com/okachobi/arduino-aaser-gateway)
